src/external_data_ifaces_impl.cpp

```diff
@@ -41,4 +42,48 @@ sdbusplus::async::task<> ExternalDataIFacesImpl::fetchBMCRedundancyMgrProps()
     co_return;
 }
 
+// NOLINTNEXTLINE
+sdbusplus::async::task<> ExternalDataIFacesImpl::fetchSiblingBmcIP()
+{
+    // TODO: Currently, the IP addresses for both BMCs are hardcoded.
+    // Once the network DBus exposes the IPs, update the logic to retrieve
+    // them dynamically from Dbus.
+
+    // TODO: Handle the exception and exit gracefully, as the data sync relies
+    //      heavily on these DBus properties and cannot function effectively
+    //      without them.
+
+    const data_sync::ext_data::SiblingBmcIP siblingIp0 = "10.0.2.100";
+    const data_sync::ext_data::SiblingBmcIP siblingIp1 = "10.2.2.100";
+
+    using siblingBmc = sdbusplus::client::xyz::openbmc_project::state::bmc::
+        redundancy::Sibling<>;
+
+    std::string object_path =
+        std::string(sdbusplus::common::xyz::openbmc_project::state::bmc::
+                        redundancy::Sibling::namespace_path::value) +
+        "/" +
+        sdbusplus::common::xyz::openbmc_project::state::bmc::redundancy::
+            Sibling::namespace_path::bmc;
+
+    auto bmcPosition = siblingBmc(_ctx)
+                           .service(
+                               sdbusplus::common::xyz::openbmc_project::state::
+                                   bmc::redundancy::Sibling::interface)
+                           .path(object_path);
+
+    auto propBmcPosition = co_await bmcPosition.bmc_position();
+
+    if (propBmcPosition == 0)
+    {
+        siblingBmcIP(siblingIp1);
```

Comment:
If the sibling BMC position is 0, we should use its BMC position, not the opposite one, right?

It seems like you may have confused the current BMC position (`fw_printenv -n bmc_position`) with the sibling BMC position (`busctl -j get-property xyz.openbmc_project.State.BMC.Redundancy.Sibling /xyz/openbmc_project/state/bmc1 xyz.openbmc_project.State.BMC.Redundancy.Sibling BMCPosition`).